agent/config/builder.go

```diff
@@ -41,57 +41,77 @@ import (
 	"github.com/hashicorp/consul/types"
 )
 
-// Load will build the configuration including the extraHead source injected
+// LoadOpts used by Load to construct and validate a RuntimeConfig.
+type LoadOpts struct {
```

Comment:
This was moved from `flags.go` and renamed to `LoadOpts` since `BuilderOpts` no longer makes sense once `Builder` is unexported.